### PR TITLE
Update rules to use a helper to declare the cpp toolchain dependency.

### PR DESCRIPTION
**Pull request by katre**

### CODE REVIEW

**Comment by katre** [comment]
The `use_cpp_toolchain` function was added in Bazel 5.1, it looks like the min version test is at Bazel 5.0.

File: rust/private/utils.bzl
Comment (UebelAndre) [inline rust/private/utils.bzl:67]: What is the benefit in using this function? It feels like this change would be better to hold until `6.0` is the min tested version.

**Review by UebelAndre — CHANGES_REQUESTED** [review]

File: rust/private/utils.bzl
Comment (katre) [reply rust/private/utils.bzl:67]: This is part of a project to make the CC toolchain optional, since many users cannot currently use toolchain resolution. It's currently causing us problems on the internal side and so I'd like to get this through, but if the rules_rust maintainers want to hold off I'll look into workarounds.

Comment (UebelAndre) [reply rust/private/utils.bzl:67]: I would prefer that `5.0` remain the min tested version as it's a current LTS release but if this is a major issue then I can concede it's not a huge deal. Though, work to support something outside of toolchain resolution feels like negative work and a burden to maintainers.

Comment (katre) [reply rust/private/utils.bzl:67]: Switching to `use_cpp_toolchain` is the currently recommended way to declare a dependency on the CC toolchain, but you're right that it's not yet in an LTS release. I'm fine with closing this PR and figuring out a workaround, but this will be a change that should be made after the next LTS release.

**Comment by katre** [comment]
This is part of a project to make the CC toolchain optional, since many
users cannot currently use toolchain resolution.

**Comment by UebelAndre** [comment]
@katre If this change is still useful, I think now would be an appropriate time to adopt it.

**Pull request closed without merging**
